src/protocols/MeterS0.cpp

```diff
@@ -238,31 +238,34 @@ void MeterS0::counter_thread()
 		if (is_blocking) {
 			bool timeout = false;
 			if (_hwif->waitForImpulse( timeout )) {
+				//Event erkannt. Wert des Events ist, aufgrund des prellen reiner Zufall!!
+				//deswegen erst entprellen .....
```

Comment:
Jetzt habe ich es geändert, wie oben disskutiert.
Laut Deinem letzten Kommentar würde hwif_dir->status jetzt aber doch den Pegel des 2. Pins liefern und hwif->status den Pegel des ersten Pin.

Dann hätte ich im neuen PR das if verschlimmbessert. 

In meinem ersten PR habe ich negative Impulse nur bei hwif_dir->status > 0 gezählt und normale Impulse bei hwif->status > 0

Jetzt zähle ich negative Impulse nur noch, wenn hwif_dir->status > 0 und hwif->status > 0 ist. Das dürfte falsch sein.